Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp

```diff
@@ -134,6 +135,46 @@ VideoFrameGStreamer::VideoFrameGStreamer(const GRefPtr<GstSample>& sample, const
 {
 }
 
+Ref<VideoFrameGStreamer> VideoFrameGStreamer::resizeTo(const IntSize& destinationSize)
+{
+    auto* caps = gst_sample_get_caps(m_sample.get());
+
+    const auto* structure = gst_caps_get_structure(caps, 0);
+    int frameRateNumerator, frameRateDenominator;
+    if (!gst_structure_get_fraction(structure, "framerate", &frameRateNumerator, &frameRateDenominator)) {
+        frameRateNumerator = 1;
+        frameRateDenominator = 1;
+    }
+
+    GstVideoInfo inputInfo;
+    gst_video_info_from_caps(&inputInfo, caps);
+    auto format = GST_VIDEO_INFO_FORMAT(&inputInfo);
+    auto width = destinationSize.width();
+    auto height = destinationSize.height();
+    auto outputCaps = adoptGRef(gst_caps_new_simple("video/x-raw", "format", G_TYPE_STRING, gst_video_format_to_string(format), "width", G_TYPE_INT, width,
+        "height", G_TYPE_INT, height, "framerate", GST_TYPE_FRACTION, frameRateNumerator, frameRateDenominator, nullptr));
+    GstVideoInfo outputInfo;
+    gst_video_info_from_caps(&outputInfo, outputCaps.get());
+
+    auto* buffer = gst_sample_get_buffer(m_sample.get());
+    auto outputBuffer = adoptGRef(gst_buffer_new_allocate(nullptr, GST_VIDEO_INFO_SIZE(&outputInfo), nullptr));
+    {
+        GUniquePtr<GstVideoConverter> converter(gst_video_converter_new(&inputInfo, &outputInfo, nullptr));
+        GstMappedFrame inputFrame(buffer, inputInfo, GST_MAP_READ);
+        GstMappedFrame outputFrame(outputBuffer.get(), outputInfo, GST_MAP_WRITE);
+        gst_video_converter_frame(converter.get(), inputFrame.get(), outputFrame.get());
+    }
+    gst_buffer_add_video_meta(outputBuffer.get(), GST_VIDEO_FRAME_FLAG_NONE, format, width, height);
```

Comment:
I don't think this meta is required, since it's useful when the strides or alignment aren't the common ones for the format.